lib/browser/api/session.ts

```diff
@@ -1,11 +1,37 @@
 import { fetchWithSession } from '@electron/internal/browser/api/net-fetch';
 import { net } from 'electron/main';
 const { fromPartition, fromPath, Session } = process._linkedBinding('electron_browser_session');
+const { isDisplayMediaSystemPickerAvailable } = process._linkedBinding('electron_browser_desktop_capturer');
+
+// Fake video source that activates the native system picker
+// This is used to get around the need for a screen/window
+// id in Chrome's desktopCapturer.
+let fakeVideoSourceId = -1;
+const systemPickerVideoSource = Object.create(null);
+Object.defineProperty(systemPickerVideoSource, 'id', {
+  get () {
+    return `window:${fakeVideoSourceId--}:0`;
+  }
+});
+systemPickerVideoSource.name = '';
+Object.freeze(systemPickerVideoSource);
```

Comment:
Makes sense and good to know chromium takes a similar approach with the magic ID.